### PR TITLE
6534 geom from site

### PR DESCRIPTION
**Pull request by serge-gaia**
Geometry for observations can be extracted from the site.

### CODE REVIEW

File: biosys/apps/main/api/serializers.py [hunk 1]
```diff
@@ -77,7 +77,9 @@ def __call__(self, data):
             validator.schema_error_as_warning = not self.strict
             result = validator.validate(data)
             if result.has_errors:
-                raise ValidationError(list(result.errors.items()))
+                error_messages = ['{col_name}::{message}'.format(col_name=k, message=v) for k, v in
+                                  result.errors.items()]
+                raise ValidationError(error_messages)
```
Comment (serge-gaia) [inline biosys/apps/main/api/serializers.py:82]: Changed the format of the errors returned by the schema validator. The client should receive a list of string with the format above.

File: biosys/apps/main/tests/api/test_observation.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
+import datetime
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/test_observation.py:1]: That's the biggie

File: biosys/apps/main/utils_data_package.py [hunk 3]
```diff
@@ -365,7 +365,7 @@ def model_field(self):
 
 
 @python_2_unicode_compatible
-class GenericSchema:
+class GenericSchema(object):
```
Comment (serge-gaia) [inline biosys/apps/main/utils_data_package.py:368]: Damn python 2! Go to hell.

File: biosys/apps/main/utils_data_package.py [hunk 4]
```diff
@@ -472,28 +472,59 @@ def get_fk_for_model(self, model_name):
...
         return self.get('name')
 
 
 class ObservationSchema(GenericSchema):
     """
-    A schema specific to an Observation Dataset.
-    It's main job is to deal with the observation date and it's geometry
-    (lat/long or geojson)
+     A schema specific to an Observation Dataset.
+     It's main job is to deal with the observation date and it's geometry
```
Comment (tony-gaia) [inline biosys/apps/main/utils_data_package.py]: No apostrophe in a possessive "its".

File: biosys/apps/main/utils_data_package.py [hunk 5]
```diff
@@ -472,28 +472,59 @@ def get_fk_for_model(self, model_name):
...
-        self.latitude_field = self.find_latitude_field_or_throw(self)
-        self.longitude_field = self.find_longitude_field_or_throw(self)
-        self.datum_field = self.find_datum_field_or_none(self)
+        site_code_fk = self.get_fk_for_model_field('Site', 'code')
+        self.site_code_field = self.get_field_by_mame(site_code_fk.data_field) if site_code_fk else None
+        if self.site_code_field is None:
+            self.latitude_field = self.find_latitude_field_or_throw(self)
+            self.longitude_field = self.find_longitude_field_or_throw(self)
+            self.datum_field = self.find_datum_field_or_none(self)
+        else:
```
Comment (tony-gaia) [inline biosys/apps/main/utils_data_package.py:521]: This seems like a convoluted way of doing things - having two different methods that either throw or don't throw. Why not just have one method but if the first case (site code is not) you throw the exception up the chain but in the second case (there is a site code) you catch the exception and do nothing with it.

**Comment by serge-gaia** [comment]
fixed comments

**Pull request merged**